src/haddock/gear/yaml2cfg.py

```diff
@@ -119,13 +119,32 @@ def _yaml2cfg_text(ycfg, module, explevel):
     return os.linesep.join(params)
 
 
-def read_from_yaml_config(cfg_file):
-    """Read config from yaml by collapsing the expert levels."""
+def read_from_yaml_config(cfg_file, default_only=True) -> dict:
+    """Read config from yaml by collapsing the expert levels.
+    
+    Parameters
+    ----------
+    cfg_file :
+        Path to a .yaml configuration file
+    default_only : bool
+        Set the return value of this function; if True (default value), only
+        returns default values, else return the fully loaded configuration file
+
+    Return
+    ------
+    ycfg : dict
+        The full default configuration file as a dict
+    OR
+    cfg : dict
+        A dictionnary containing only the default parameters values
```

Comment:
is there any particular reason to have two different dict objects in output?

Reply:
There is the original one `cfg` that only contains the ['default'] value and used in other parts of the code.
While the `ycfg` one contrains verything present in the default.yaml file loaded as  a dict, used for the analysis of the agreement (type, range, choices, sizes) between user provided configuration and authorized one.